pkgs/by-name/we/weblate/package.nix

```diff
@@ -0,0 +1,187 @@
+{
+  lib,
+  python3,
+  fetchFromGitHub,
+  fetchpatch,
+  pango,
+  harfbuzz,
+  librsvg,
+  gdk-pixbuf,
+  glib,
+  borgbackup,
+  writeText,
+  nixosTests,
+}:
+
+let
+  python = python3.override {
+    packageOverrides = final: prev: {
+      django = prev.django_5.overridePythonAttrs (old: {
+        dependencies = old.dependencies ++ prev.django_5.optional-dependencies.argon2;
+      });
+      sentry-sdk = prev.sentry-sdk_2;
+      djangorestframework = prev.djangorestframework.overridePythonAttrs (old: {
+        # https://github.com/encode/django-rest-framework/discussions/9342
+        disabledTests = (old.disabledTests or [ ]) ++ [ "test_invalid_inputs" ];
+      });
+      celery = prev.celery.overridePythonAttrs (old: {
+        dependencies = old.dependencies ++ prev.celery.optional-dependencies.redis;
+      });
+      python-redis-lock = prev.python-redis-lock.overridePythonAttrs (old: {
+        dependencies = old.dependencies ++ prev.python-redis-lock.optional-dependencies.django;
+      });
+    };
+  };
+in
+python.pkgs.buildPythonApplication rec {
+  pname = "weblate";
+  version = "5.6.2";
+
+  pyproject = true;
+
+  outputs = [
+    "out"
+    "static"
+  ];
+
+  src = fetchFromGitHub {
+    owner = "WeblateOrg";
+    repo = "weblate";
+    rev = "weblate-${version}";
+    sha256 = "sha256-t/hnigsKjdWCkUd8acNWhYVFmZ7oGn74+12347MkFgM=";
+  };
+
+  patches = [
+    # FIXME This shouldn't be necessary and probably has to do with some dependency mismatch.
+    ./cache.lock.patch
+    # relax dependency constraint for borgbackup
+    (fetchpatch {
+      url = "https://github.com/WeblateOrg/weblate/commit/355c81c977c59948535a98a35a5c05d7e6909703.patch";
+      sha256 = "sha256-9cgE+KeSsXweb2JUvstMxEehfNVktZm1geAo496EPN0=";
+    })
+    # relax dependency constraint for django-crispy-forms
+    (fetchpatch {
+      url = "https://github.com/WeblateOrg/weblate/commit/7b341c523ed9b3b41ecfbc5c92dd6156992e4f32.patch";
+      sha256 = "sha256-AcWQD+Z1wGX9MT3IFiomN4sncsQ2VOtflo6BLRZbxBc=";
+    })
+  ];
+
+  # relax dependency constraint for mistletoe
+  # https://github.com/WeblateOrg/weblate/commit/50df46a25dda2b7b39de86d4c65ecd7a685f62e6
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace '"mistletoe>=1.3.0,<1.4"' '"mistletoe>=1.3.0,<1.5"'
+  '';
```

Comment:
Is there a reason mistletoe is being relaxed in a different way than borgbackup and django-crispy-forms? Does that patch not apply?

Maybe we can switch them all to the `substituteInPlace` method. It makes it clear what's being updated, and makes it easier to update in the future. Do we need to be careful to only update things that have been updated upstream? Does that even guarantee us anything, given that something might have changed upstream in the meantime to add compatibility with that version?

Reply:
Yeah, the specific patch didn't apply. I liked the idea of having an easy way to verify that the change was also done by upstream, but ofc it doesn't guarantee anything as you said. I'll switch to `substituteInPlace` and add comments to the respective commits.